indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -3183,35 +3181,34 @@ private void checkTaskDuration() throws ExecutionException, InterruptedException
           Integer groupId = entry.getKey();
           TaskGroup group = entry.getValue();
 
-          if (stopTasksEarly) {
+          final DateTime earliestTaskStart = computeEarliestTaskStartTime(group);
+          final Duration runDuration = Duration.millis(DateTimes.nowUtc().getMillis() - earliestTaskStart.getMillis());
+          if (stopTasksEarly || group.getHandoffEarly()) {
+            // If shutdownEarly has been set, stop tasks irrespective of stopTaskCount
```

Comment:
Change to `If getHandoffEarly has been set ....`

Reply:
done.